src/services/turbo/TurboArNSClient.ts

```diff
@@ -281,9 +322,45 @@ export class TurboArNSClient {
     if (result.error) {
       throw new Error(result.error.message);
     }
+
+    const maxTries = 10;
+    let tries = 0;
+    let isComplete = false;
+    let messageId: string | undefined;
+    while (!isComplete && tries <= maxTries) {
+      const res = await this.getIntentStatus(intent.purchaseQuote.nonce);
+      switch (res.status) {
+        case 'success':
+          isComplete = true;
+
+          messageId = res.messageId;
+
+          break;
+        case 'failed':
+          throw new Error('Turbo ArNS Interaction failed on payment service.');
+        case 'pending':
+          if (tries >= maxTries) {
+            throw new Error(
+              'Turbo ArNS Interaction exceeded max tries on payment service.',
+            );
+          }
+          await sleep(1000 * tries);
+
+          tries++;
+          break;
+        default:
+          throw new Error('Unknown status from payment service.');
+      }
+    }
+    if (!messageId) {
+      throw new Error('No message ID found on payment service.');
+    }
     return {
-      paymentResult: result,
-      paymentIntent: intent.paymentSession,
+      id: messageId,
+      result: (await this.ao.result({
```

Comment:
nit: odd to await here on the response object vs above

Reply:
and why do we need to cast?

Reply:
I was borking the return type, fixed. also awaiting above. https://github.com/ar-io/arns-react/pull/753/commits/21d18a0156991f8d510c641606d4532a4950eac1